core/domain/rights_manager.py

```diff
@@ -382,7 +384,23 @@ def create_new_collection_rights(collection_id, committer_id):
     subscription_services.subscribe_to_collection(committer_id, collection_id)
 
 
-def get_collection_rights(collection_id, strict=True):
+@overload
+def get_collection_rights(
+    collection_id: str
+) -> rights_domain.ActivityRights: ...
+
+
+@overload
+def get_collection_rights(
+    collection_id: str,
+    strict: bool = False
+) -> Optional[rights_domain.ActivityRights]: ...
+
+
+def get_collection_rights(
+    collection_id: str,
+    strict: bool = True
+) -> Optional[rights_domain.ActivityRights]:
```

Comment:
This is still weird to me, because if we explicitly set `strict` to `True` the type is not right, right? Is it possible to rewrite this so that if we explicitly mention strict then we don't get `Optional`?

Reply:
Done!
Now:
```
get(aaa, strict=True) ------------------ will not return Optional
get(aaa, strict=False) ----------------- will return optional
```
Also, overloads that contain strict as a positional argument is removed.

Reply:
Hmm, sorry I am a bit confused by how you handle these. To my understanding, I would imagine something like this. This both allows the call without strict and enforces string to be only used by keyword.

```python
@overload
def get_collection_rights(
    collection_id: str
) -> rights_domain.ActivityRights: ...


@overload
def get_collection_rights(
    collection_id: str, *, strict: Literal[True]
) -> rights_domain.ActivityRights: ...


@overload
def get_collection_rights(
    collection_id: str, *, strict: Literal[False]
) -> Optional[rights_domain.ActivityRights]: ...


def get_collection_rights(
    collection_id: str, *, strict: bool = True
) -> Optional[rights_domain.ActivityRights]:
```

ditto elsewhere

Reply:
Done wherever it is applicable!